contracts/RelayHub.sol

```diff
@@ -292,41 +303,53 @@ contract RelayHub is IRelayHub {
             // acceptRelayedCall that this will not happen.
             (success, retData) = relayRequest.relayData.paymaster.call{gas:gasLimits.preRelayedCallGasLimit}(atomicData.data);
             if (!success) {
-                revertWithStatus(RelayCallStatus.PreRelayedFailed, GsnEip712Library.getTruncatedData(retData));
+                revertWithStatus(RelayCallStatus.PreRelayedFailed, retData);
             }
-            atomicData.preReturnValue = abi.decode(retData, (bytes32));
+            (atomicData.recipientContext, atomicData.isTrustedRecipient) = abi.decode(retData, (bytes,bool));
```

Comment:
Maybe rename to revertOnRecipientRevert? since that's what distinguishes between "pre" mode and "recipient" mode per the doc

Reply:
renamed.